Mathlib/RingTheory/MvPowerSeries/Topology.lean

```diff
@@ -0,0 +1,293 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos Fernández
+-/
+
+import Mathlib.RingTheory.Nilpotent.Defs
+import Mathlib.RingTheory.MvPowerSeries.Basic
+import Mathlib.Topology.Algebra.InfiniteSum.Constructions
+import Mathlib.Topology.Algebra.Ring.Basic
+import Mathlib.Topology.Algebra.UniformGroup
+import Mathlib.Topology.UniformSpace.Pi
+
+/-! # Topology on power series
+
+Let `R` be with Semiring R` and `TopologicalSpace R`
+In this file we the topology on `MvPowerSeries σ R`
+that corresponds to the simple convergence on its coefficients.
+It is the coarsest topology for which all coefficients maps are continuous.
+
+When `R` has `UniformSpace R`, we define the corresponding uniform structure.
+
+When the type of coefficients has the discrete topology,
+it corresponds to the topology defined by [bourbaki1981], chapter 4, §4, n°2
+
+It is *not* the adic topology in general.
+
+- `tendsto_pow_zero_of_constantCoeff_nilpotent`, `tendsto_pow_zero_of_constantCoeff_zero`:
+if the constant coefficient of `f` is nilpotent, or vanishes,
+then the powers of `f` converge to zero.
+
+- `tendsto_pow_of_constantCoeff_nilpotent_iff` : the powers of `f` converge to zero iff
+the constant coefficient of `f` is nilpotent
+
+- `hasSum_of_monomials_self` : viewed as an infinite sum, a power series coverges to itself
+
+TODO: add the similar result for the series of homogeneous components
+
+## Instances
+
+- If `R` is a topological (semi)ring, then so is `MvPowerSeries σ R`
+
+- If the topology of `R` is T2, then so is that of `MvPowerSeries σ R``
+
+- If `R` is a `uniformAddGroup`, then so is `MvPowerSeries σ R``
+
+- If `R` is complete, then so is `MvPowerSeries σ R`
+
+-/
+
+
+theorem Finset.prod_one_add {ι R : Type _} [DecidableEq ι] [CommRing R] {f : ι → R} (s : Finset ι) :
```

Comment:
Please move this to a better home.

If you write `#find_home! Finset.prod_one_add` below it, then Lean will give you some candidates. (Hopefully good ones.)